src/main/java/frc/utils/AutonomousSelector.java

```diff
@@ -0,0 +1,31 @@
+package frc.utils;
+
+import com.pathplanner.lib.auto.AutoBuilder;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import org.littletonrobotics.junction.Logger;
+
+public class AutonomousSelector {
+
+    private static SendableChooser<Command> selector;
+
+    public static void buildSelector() {
+        selector = AutoBuilder.buildAutoChooser();
+        ShuffleboardTab tab = Shuffleboard.getTab("auto");
+        tab.add("autonomous selector", selector);
+    }
+
+    public static Command getChosenValue() {
+        if (selector != null) {
+            Logger.recordMetadata("autonomous chosen", selector.getSelected().getName());
+            return selector.getSelected();
+        }
+        else {
+            Logger.recordMetadata("autonomous chosen", "None");
+            return new InstantCommand();
+        }
```

Comment:
I know i suggested static but consider change it again to not static, also it would remove this if else block.

Reply:
now that the selector is final there's no need for the block.
I think the static implementation is cleaner because we only need one, and I see the selector as a utility to access its data

Reply:
I can move the selector to a final in robot container and make the class an actual util if you want